src/Testcontainers.DockerCompose/DockerComposeBuilder.cs

```diff
@@ -34,68 +33,52 @@ private DockerComposeBuilder(DockerComposeConfiguration resourceConfiguration)
+            .WithResourceMapping(new FileInfo(composeFilePath), new FileInfo(DockerComposeFilePath));
     }
 
     /// <summary>
-    ///     If true use a local Docker Compose binary instead of a container.
+    /// Sets the Docker Compose mode.
     /// </summary>
-    /// <param name="localCompose">Whether the local compose will be used.</param>
+    /// <param name="mode">The Docker Compose mode.</param>
     /// <returns>A configured instance of <see cref="DockerComposeBuilder" />.</returns>
-    public DockerComposeBuilder WithLocalCompose(bool localCompose)
+    public DockerComposeBuilder WithComposeMode(DockerComposeMode mode)
     {
-        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration
-            (localCompose: localCompose));
+        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration(mode: mode));
     }
 
-    /// <summary>
-    ///     Adds options to the docker-compose command, e.g. docker-compose --compatibility.
-    /// </summary>
-    /// <param name="options">Options for the docker-compose command.</param>
-    /// <returns>A configured instance of <see cref="DockerComposeBuilder" />.</returns>
-    public DockerComposeBuilder WithOptions(params string[] options)
+    /// <inheritdoc />
+    public override DockerComposeContainer Build()
     {
-        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration(options: options))
-            .WithCommand(options);
-    }
+        Validate();
 
-    /// <summary>
-    ///     Remove images after containers shutdown.
-    /// </summary>
-    /// <param name="removeImages"></param>
-    /// <returns>A configured instance of <see cref="DockerComposeBuilder" />.</returns>
-    public DockerComposeBuilder WithRemoveImages(RemoveImages removeImages)
-    {
-        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration(removeImages: removeImages));
+        switch (DockerResourceConfiguration.Mode)
+        {
+            case DockerComposeMode.Local:
+                return new DockerComposeLocalContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+            case DockerComposeMode.Remote:
+                return new DockerComposeRemoteContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+            default:
+                throw new ArgumentOutOfRangeException(nameof(DockerResourceConfiguration.Mode), "Docker Compose mode not supported.");
+        }
     }
-        
+
     /// <inheritdoc />
     protected override DockerComposeBuilder Init()
     {
         return base.Init()
             .WithImage(DockerComposeImage)
-            .WithEntrypoint(DockerComposeCommandLineBuilder.DockerAppName)
-            .WithCommand(DockerComposeCommandLineBuilder
-                .FromRemoteConfiguration(DockerResourceConfiguration)
-                .BuildStartCommand()
-                .ToArray())
+            .WithCommand("up")
```

Comment:
How do we know the container is up and running ([ready](https://dotnet.testcontainers.org/api/wait_strategies/))? How do we know that the services from the Compose file are ready and accepting connections, etc.?